services/tables/src/main/java/com/linkedin/openhouse/tables/utils/TableUUIDGenerator.java

```diff
@@ -64,12 +66,44 @@ public UUID generateUUID(CreateUpdateTableRequestBody createUpdateTableRequestBo
   public UUID generateUUID(IcebergSnapshotsRequestBody icebergSnapshotsRequestBody) {
     return extractUUIDFromSnapshotJson(
             icebergSnapshotsRequestBody.getJsonSnapshots(),
-            icebergSnapshotsRequestBody.getCreateUpdateTableRequestBody().getDatabaseId(),
-            icebergSnapshotsRequestBody.getCreateUpdateTableRequestBody().getTableId())
+            extractFromTblPropsIfExists(
```

Comment:
we need to change the signature of `extractUUIDFromSnapshotJson(` to  `extractUUIDFromSnapshotJson( List<String> jsonSnapshots, Map<String, String> tblProps)`. The old signature doesn't make sense anymore with the new revelations.